WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift

```diff
@@ -218,16 +218,18 @@ import WordPressComAnalytics
         let service = ReaderTopicService(managedObjectContext: ContextManager.sharedInstance().mainContext)
         service.siteTopicForSiteWithID(siteID!,
             isFeed:isFeed,
-            success: { [weak self] (objectID:NSManagedObjectID!, isFollowing:Bool) -> Void in
-                do {
-                    let context = ContextManager.sharedInstance().mainContext
-                    let topic = try context.existingObjectWithID(objectID) as? ReaderAbstractTopic
-                    self?.readerTopic = topic
-                } catch let error as NSError {
-                    DDLogSwift.logError(error.localizedDescription)
+            success: { [weak self] (objectID:NSManagedObjectID!, isFollowing:Bool) in
+
+                let context = ContextManager.sharedInstance().mainContext
+                guard let topic = (try? context.existingObjectWithID(objectID)) as? ReaderAbstractTopic else {
+                    DDLogSwift.logError("Reader: Error retriving an existing site topic by its objectID")
+                    self?.displayLoadingStreamFailed()
+                    return
                 }
+                self?.readerTopic = topic
+
             },
-            failure: {[weak self] (error:NSError!) -> Void in
+            failure: { [weak self] (error:NSError!) in
```

Comment:
Same as above, because this error isn't being used:

`failure: { [weak self] _ in`

I think there might be quite a few places this could be done in here.